src/main/java/org/mule/runtime/dsl/internal/xml/parser/XmlMetadataAnnotations.java

```diff
@@ -16,6 +16,49 @@
 @NoImplement
 public interface XmlMetadataAnnotations {
 
+  interface TagBoundaries {
```

Comment:
Should I move this to a top level interface instead of having it nested here?

Reply:
I have no concerns with this being nested, it is just a matter of code readabality. Since it is internal, it can be extracted if needed in the future without breaking anything.